.github/guides/STYLE.md

```diff
@@ -399,6 +400,328 @@ turn_on(power_usage = 30) // Fine!
+
+```
+// Guarantee 
+#define CALCULATE_TEMPERATURE(base) (T20C + (##base))
+```
+
+### Be hygienic where reasonably possible
+
+Consider the previously mentioned non-hygienic macro:
+
+```dm
+#define GET_HEALTH(health_percent) (##health_percent * max_health)
+```
+
+This relies on "max_health", but it is not obviously clear what the source is. This will also become worse if we *do* want to change where we get the source from. This would be preferential as:
+
+```dm
+#define GET_HEALTH(source, health_percent) (##health_percent * ##source.max_health)
+```
+
+When a macro can't be hygienic, such as in the case where a macro is preferred to do something like define a variable, it should still do its best to rely only on input given to it:
+
+```dm
+#define DECLARE_MOTH(name) var/mob/living/moth/moth = new(##name)
+```
+
+...would ideally be written as...
+
+```dm
+#define DECLARE_MOTH(var_name, name) var/mob/living/moth/##var_name = new(##name)
+```
+
+As usual, exceptions exist--for instance, accessing a global like a subsystem within a macro is generally acceptable.
+
+### Preserve hygiene using double underscores (`__`) and `do...while (FALSE)`
+
+Some macros will want to create variables for themselves, and not the consumer. For instance, consider this macro:
+
+```dm
+#define HOW_LONG(proc_to_call) \
+	var/current_time = world.time; \
+	##proc_to_call(); \
+	world.log << "That took [world.time - current_time] deciseconds to complete.";
+```
+
+There are two problems here.
+
+One is that it is unhygienic. The `current_time` variable is leaking into the call site.
+
+The second is that this will create weird errors if `current_time` is a variable that already exists, for instance:
+
+```dm
+var/current_time = world.time
+
+HOW_LONG(make_soup) // This will error!
+```
+
+If this seems unlikely to you, then also consider that this:
+
+```dm
+HOW_LONG(make_soup)
+HOW_LONG(eat_soup)
+```
+
+...will also error, since they are both declaring the same variable!
+
+There is a way to solve both of these, and it is through both the `do...while (FALSE)` pattern and by using `__` for variable names.
+
+This code would change to look like:
+
+```dm
+#define HOW_LONG(proc_to_call) \
+	do { \
+		var/__current_time = world.time; \
+		##proc_to_call(); \
+		world.log << "That took [world.time - current_time] deciseconds to complete."; \
+	} while (FALSE)
+```
+
+The point of the `do...while (FALSE)` here is to **create another scope**. It is impossible for `__current_time` to be used outside of the define itself. If you haven't seen `do...while` syntax before, it is just saying "do this while the condition is true", and by passing `FALSE`, we ensure it will only run once.
+
+### Keep anything you use more than once in variables
+
+Remember that macros are just pastes. This means that, if you're not thinking, you can end up [creating some really weird macros by reusing variables](https://github.com/tgstation/tgstation/pull/55074).
```

Comment:
I may be dumb but what's the relevance of this pr link?

Reply:
Check files changed--it's the fix for the bug that is being explained in this article

Reply:
OHHHH I was looking at https://github.com/tgstation/tgstation/pull/5507